hollow/src/main/java/com/netflix/hollow/api/producer/HollowProducer.java

```diff
@@ -630,15 +640,18 @@ private void applyDelta(Blob blob, HollowReadStateEngine stateEngine) throws IOE
 
     private void validate(HollowProducer.ReadState readState) {
         ProducerStatus.Builder status = listeners.fireValidationStart(readState);
+        ValidationStatus.Builder valStatus = ValidationStatus.builder().withReadState(readState).withVersion(readState.getVersion());
         try {
             List<Throwable> validationFailures = new ArrayList<Throwable>();
             
             for(Validator validator : validators) {
+            	Throwable throwable = null;
                 try {
                     validator.validate(readState);
```

Comment:
Validate.validate interface does not seem to indicate it should throw exception on failure.   Might be good idea to make it clearer.

Reply:
That is true. That is also being considered for next iteration.